snuba/query/processors/arrayjoin_keyvalue_optimizer.py

```diff
@@ -80,7 +80,7 @@ def _get_mapping_keys_in_condition(
     return keys_found
 
 
-def get_filtered_mapping_keys(query: Query, column_name: str) -> Set[str]:
+def get_filtered_mapping_keys(query: Query, column_name: str) -> List[str]:
```

Comment:
I think you should be able to return a Sequence so we are sure nobody adds something by mistake (before it would not be an issue as there could not be duplicates as it was a set).